pkg/prometheus/operator.go

```diff
@@ -127,14 +127,22 @@ func ValidateRemoteWriteSpec(spec monitoringv1.RemoteWriteSpec) error {
 	}
 
 	if spec.AzureAD != nil {
-		if spec.AzureAD.ManagedIdentity == nil && spec.AzureAD.OAuth == nil {
-			return fmt.Errorf("must provide Azure Managed Identity or Azure OAuth in the Azure AD config")
+		if spec.AzureAD.ManagedIdentity == nil && spec.AzureAD.OAuth == nil && spec.AzureAD.SDK == nil {
```

Comment:
There's still a risk that users pick up the SDK option with an older Prometheus version in which the generated config will fail. Ideally the operator should fail here.

Reply:
should we have this validated similar to relabel configs is resource_selector.go?